script.rb

```diff
@@ -0,0 +1,184 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    length.times do
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    length.times do
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    i = 0
+    my_new_arr = [] # It will return a new array
+    length.times do
+      value = yield self[i]
+      my_new_arr.push self[i] unless value == true
+      i += 1
+    end
+    my_new_arr
+  end
```

Comment:
- [ ] my_select should return an array containing all elements of enum for which the given block returns a true value when self is an array or a range. Currently, my_select does not behave as it is expected. As stated above the best way to test your method is to test it against the original method and make sure they both have the same behavior.